### DISCUSSION TITLE
How to obtain the force and velocity of a collision

### DISCUSSION THREAD

**Discussion by taosimple** [flame-engine/flame#3328] [Q&A]
There are many small balls in a container. I want to add a vibration effect when the small balls collide. However, when a collision occurs, `beginContact` or `endContact` will be called a large number of times. I want to add vibration only to the collisions with greater impact and ignore those with smaller impact. But I have tested many properties and consulted the documentation and haven't found a correct way to obtain the impact force and speed.

remarks: use flame_forge2d
<img width="218" alt="image" src="https://github.com/user-attachments/assets/8c6a9532-909a-4240-b807-94eb74287653">

**Comment by spydon** [comment]
You should be able to override `postSolve` from the `ContactsCallbacks` mixin with something like this:
```dart
  @override
  void postSolve(Object other, Contact contact, ContactImpulse impulse) {
    // Calculate the collision force
    final normalImpulse = impulse.normalImpulses[0];
    final tangentImpulse = impulse.tangentImpulses[0];

    // Assuming the mass of the bodies is available
    final bodyA = contact.fixtureA.body;
    final bodyB = contact.fixtureB.body;

    final massA = bodyA.mass;
    final massB = bodyB.mass;

    // Calculate the force using the impulse and mass
    final forceA = normalImpulse / massA;
    final forceB = normalImpulse / massB;

    print('Collision force on bodyA: $forceA');
    print('Collision force on bodyB: $forceB');
  }
```
